flocker/apiclient/_client.py

```diff
@@ -144,4 +150,127 @@ def synchronize_state(self):
+
+
+@implementer(IFlockerAPIV1Client)
+class FlockerClient(object):
+    """
+    A client for the Flocker V1 REST API.
+    """
+    def __init__(self, reactor, host, port,
+                 ca_cluster_path, cert_path, key_path):
+        """
+        :param reactor: Reactor to use for connections.
+        :param bytes host: Host to connect to.
+        :param int port: Port to connect to:
+        :param FilePath ca_cluster_path: Path to cluster's CA certificate.
+        :param FilePath cert_path: Path to user certificate.
+        :param FilePath key_path: Path to user private key.
+        """
+        self._treq = treq_with_authentication(reactor, ca_cluster_path,
+                                              cert_path, key_path)
+        self._base_url = b"https://{}:{}/v1".format(host, port)
+
+    def _request(self, method, path, body, success_code, error_codes={}):
+        """
+        Send a HTTP request to the Flocker API, return decoded JSON body.
+
+        :param bytes method: HTTP method, e.g. PUT.
+        :param bytes path: Path to add to base URL.
+        :param body: If not ``None``, JSON encode this and send as the
+            body of the request.
+        :param int success_code: Expected success response code.
+        :param error_codes: Mapping from HTTP response code to exception to be
+            raised if it is present.
+
+        :return: ``Deferred`` firing with decoded JSON.
+        """
+        def error(body, code):
+            if code in error_codes:
+                raise error_codes[code](body)
+            raise ResponseError(code, body)
+
+        def got_result(result):
+            if result.code == success_code:
+                return json_content(result)
+            else:
+                d = content(result)
+                d.addCallback(error, result.code)
+                return d
+
+        headers = None
+        data = None
+        if body is not None:
+            headers = {b"content-type": b"application/json"}
+            data = dumps(body)
+        request = self._treq.request(
+            method, self._base_url + path,
+            data=data, headers=headers,
+            # Keep tests from having dirty reactor problems:
+            persistent=False
+        )
+        request.addCallback(got_result)
+        return request
+
+    def _parse_configuration_dataset(self, dataset_dict):
+        """
+        Convert a dictionary decoded from JSON with a dataset's configuration.
+
+        :param dataset_dict: Dictionary describing a dataset.
+        :return: ``Dataset`` instance.
+        """
+        return Dataset(primary=UUID(dataset_dict[u"primary"]),
+                       maximum_size=dataset_dict[u"maximum_size"],
+                       dataset_id=UUID(dataset_dict[u"dataset_id"]),
+                       metadata=dataset_dict[u"metadata"])
+
+    def create_dataset(self, primary, maximum_size, dataset_id=None,
+                       metadata=pmap()):
```

Comment:
It will, but that's kinda bogus picked-out-of-a-hat number. Better to force the caller to pick a number out of a hat, as the Docker plugin does :wink: